src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,40 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private static final StorageDao storageDao = new StorageDaoImpl();
+    private static final int MIN_USER_AGE = 18;
+    private static final int MIN_PASSWORD_CHARACTERS = 6;
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can`t  be null");
+        }
+
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password can`t be null");
+        }
+        if (user.getPassword().length() < MIN_PASSWORD_CHARACTERS) {
+            throw new RegistrationException("Not valid password!");
+        }
+
+        if (user.getAge() != null && user.getAge() < MIN_USER_AGE) {
+            throw new RegistrationException(" Not valid age!");
+        }
+
+        if (user.getAge() == null) {
+            throw new RegistrationException("Age can`t be null!");
+        }
```

Comment:
And make exception messages more informative
```suggestion
        if (user.getAge() == null) {
            throw new RegistrationException("Age can`t be null!");
        }
        if (user.getAge() < MIN_USER_AGE) {
            throw new RegistrationException(" Not valid age!");
        }
```